### ISSUE TITLE
tests: driver:eeprom: frdm_mcxn947/mcxn947/cpu0: tests/drivers/build_all/eeprom: build failure

### ISSUE THREAD

**Issue by hakehuang** [zephyrproject-rtos/zephyr#79761]
**Describe the bug**

build failure for tests/drivers/build_all/eeprom

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - frdm_mcxn947/mcxn947/cpu0
 - What have you tried to diagnose or workaround this issue?
 - NA
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
- No
 
this issue is exposed by #23496f52087c44ea06df1045d25cc4c95c24c6e8 

**To Reproduce**

scripts/twister -p frdm_mcxn947/mcxn947/cpu0 -T tests/drivers/build_all/eeprom --build-only

**Expected behavior**

Build PASS

**Impact**

eeprom support for frdm_mcxn947/mcxn947/cpu0

**Logs and console output**

```
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c: In function 'mcux_lpcmp_sample_fetch':
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c:276:25: error: 'val' undeclared (first use in this function)
        __ASSERT_NO_MSG(val != NULL);
                        ^~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/__assert.h:113:23: note: in definition of macro '__ASSERT_NO_MSG'
                 if (!(test)) {                                            \
                       ^~~~
/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c:276:25: note: each undeclared identifier is reported only once for each function it appears in
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux)
 - Toolchain (e.g Zephyr SDK, ...): 
 - Commit SHA or Version used: v3.7.0-4339-g1ec5ce05f9f8

**Comment by henrikbrixandersen** [comment]
Duplicate of #79760? Should be fixed by https://github.com/zephyrproject-rtos/zephyr/pull/79742
